cEP-0016.md

```diff
@@ -0,0 +1,332 @@
+
+        @abc.abstractmethod
+        def get_head_commit(self):
+            """
+            Return the commit message from the head commit
+            """
+
+        def analyze_git_commit(self, stdout):
+            commit_info = dict()
+
+            commit_sha = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+
+            commit_type = []
+
+            stdout = stdout.strip('\n')
+
+            if 'ci skip' in stdout or 'skip ci' in stdout:
+                commit_type.append('ci_skip_commit')
+
+            get_parent_commits = 'git log --pretty=%P -n 1 ' + commit_sha
+            parent_commits, _ = run_shell_command(get_parent_commits)
+            parent_commits_list = parent_commits.split('\n')
+
+            if len(parent_commits_list) > 2:
+                commit_type.append('merge_commit')
+
+            get_modified_files = 'git show --pretty="" --name-only ' + commit_sha
+            modified_files, _ = run_shell_command(get_modified_files)
+            modified_files_list = modified_files.split('\n')
+
+            commit_info['commit_sha'] = commit_sha
+            commit_info['commit_type'] = commit_type
+            commit_info['modified_files'] = modified_files_list
+
+            commit_info['commit_type'] = commit_type
+
+            return str(commit_info)
+
+        def run(self):
+            (stdout, stderr) = self.get_head_commit()
+
+            if stderr:
+                vcs_name = list(self.LANGUAGES)[0].lower()+':'
+                self.err(vcs_name, repr(stderr))
+                return
+
+            commit_info = self.analyze_git_commit(stdout)
+
+            yield CommitResult(self, (commit_info,), 'HEAD commit information')
+```
+
+A prototype of `SkipCIBear` :
+
+```python
+    # import statements
+
+    class SkipCIBear(GlobalBear):
+
+        BEAR_DEPS = {VCSCommitBear}
+
+        CI_PROVIDERS_METADATA = {
+          'circleci': r'/ci skip|skip ci/',
+          'travisci': r'/ci skip|skip ci/',
+          'appveyor': None
+        }
+
+        def run(self, dependency_results, allow_skipping_ci_build: bool = True):
+            if not allow_skipping_ci_build:
+                result_message = ('Skipping CI build is not allowed.')
+
+            result = dependency_results[VCSCommitBear.name]
+            commit_info = result.contents
+            # convert string representation of dictionary into dictionary
+
+            if not 'ci_skip_commit' in commit_info['commit_type']:
+                return
+
+            commit_message = 'git log --format=%B -n 1 ' + commit_info['commit_sha']
+
+            for ci_engine in self.CI_PROVIDERS_METADATA:
+                supported_regex = self.CI_PROVIDERS_METADATA[ci_engine]
+                re.compile(supported_regex)
+
+                if not match:
+                    return
+
+                if (not supported_regex):
+                    result_message = (
+                        '%s does not allow skipping CI builds.' % ci_engine)
```

Comment:
this is going to cause problems.

if a PR builds on both Circle and Appveyor, and someone uses `[ci skip]` and it is permitted, this will cause a failure on AppVeyor and not a failure on Circle CI.

which will cause every PR to fail.

If the repo wanted very `[ci skip]` to fail, they would simply use `allow_skipping_ci_build`.

Rather than try to find a solution for this problem now, I suggest that you remove this check + message, and focus on the `modified_files` check below which is looking quite good as an illustration of the bear.

Reply:
Okay. Removing this block of check and just checking `modified_files` for now.